File: src/Fundamentals.Lang.CSharp/ConditionalOperators/IConditionalOperator.cs
// <copyright file="IConditionalOperator.cs" company="Andrey Pudov">
//     Copyright (c) Andrey Pudov. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE.txt in the project root for license information.
// </copyright>

namespace Fundamentals.Lang.CSharp.ConditionalOperators
{
    /// <summary>
    /// Exposes an operator that provides the conditional execution.
    /// </summary>
    public interface IConditionalOperator
    {
        /// <summary>
        /// Returns the name of the given color.
        /// </summary>
        /// <param name="color">The instance of the color.</param>
        /// <returns>The name of the given color.</returns>
        string GetName(Color color);
    }
}

File: src/Fundamentals.Lang.FSharp/ConditionalOperators/IConditionalOperator.fs
// <copyright file="IConditionalOperator.fs" company="Andrey Pudov">
//     Copyright (c) Andrey Pudov. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE.txt in the project root for license information.
// </copyright>

namespace Fundamentals.Lang.FSharp.ConditionalOperators

open Fundamentals.Lang.FSharp

type IConditionalOperator =
    abstract member GetName : Color -> string